src/transport_performance/urban_centres/raster_uc.py

```diff
@@ -317,13 +300,8 @@ def _cluster_cells(
             Number of clusters identified.
 
         """
-        if not isinstance(flag_array, np.ndarray):
-            raise TypeError(
-                "`flag_array` expected numpy array, "
-                f"got {type(flag_array).__name__}."
-            )
-        if not isinstance(diag, bool):
-            raise TypeError("`diag` must be a boolean.")
+        d._type_defence(flag_array, "flag_array", np.ndarray)
+        d._type_defence(diag, "diag", bool)
 
         if diag is False:
```

Comment:
Changed as suggested. Since there's a defence that would not accept anything other than a `bool`, there's no risk of `else` being anything else than `False`.